### ISSUE TITLE
index out of bound exception with some names

### ISSUE THREAD

**Issue by abdullahalmaz** [eldersantos/phonix#5]
I'm getting this error with a few names, mostly those which end with an "i".
For example the name "marggi".

`var _generator = new MatchRatingApproach();`
 `Console.WriteLine(_generator.MatchRatingCompute("hofer", "marggi"));`

I'm using VS2017 with .NET Framework 4.6.1.

Could that be fixed?

**Comment by abdullahalmaz** [comment]
I found the Problem.
The Exception happens in the MatchRatingApproach class in the MatchRatingCompute function in the following loop:
```
for (int i = 0; i < small.Length;)
{
	bool found = false;
	for (int j = 0; j < large.Length; j++)
	{
		if (small[i] == large[j])
		{
			small = small.Remove(i, 1);
			large = large.Remove(j, 1);
			found = true;
		}
	}

	if (!found)
	{
		i++;
	}
}
```

If the shorter name is exactly 1 character shorter than the long name and the end character of the shorter name is contained in the long name than we have an exception.

In my case, when `small[i] == large[j]` is true, the "r" of "hofer" will be removed so it is "hofe" and in the next run of the loop small[4] doenst exist, because it has been removed, so "index out of bound exception".

By putting a "try & catch" inside the for-loop the problem is solved for me, but there should be a cleaner way to solve this.
